juju/model.py

```diff
@@ -231,6 +289,54 @@ class RelationNotFound(ModelError):
     pass
 
 
+class Status:
+    """Status values specific to applications and units."""
+    def __init__(self, id, message):
+        self.id = id
+        self.message = message
+
+class Active(Status):
+    """The unit believes it is correctly offering all the services it has been asked to offer."""
+    def __init__(self, message):
+        super().__init__('active', message)
+
+class Blocked(Status):
+    """The unit needs manual intervention to get back to the Running state."""
+    def __init__(self, message):
+        super().__init__('blocked', message)
+
+class Maintenance(Status):
+    """
+    The unit is not yet providing services, but is actively doing work in preparation for providing those services.
+    This is a "spinning" state, not an error state. It reflects activity on the unit itself, not on peers or related units.
+    """
+    def __init__(self, message):
+        super().__init__('maintenance', message)
+
+class Unknown(Status):
+    """A unit-agent has finished calling install, config-changed and start, but the charm has not called status-set yet."""
+    def __init__(self, message=''):
+        # Unknown status cannot be set and does not have a message associated with it.
+        super().__init__('unknown', '')
+
+class Waiting(Status):
+    """The unit is unable to progress to an active state because an application to which it is related is not running."""
+    def __init__(self, message):
+        super().__init__('waiting', message)
+
+@unique
+class StatusTypes(Enum):
+    active = Active
+    blocked = Blocked
+    maintenance = Maintenance
+    unknown = Unknown
+    waiting = Waiting
```

Comment:
That seems bogus. This is an example of Enum usage, from the PEP:
```
>>> from enum import Enum
>>> class Color(Enum):
...     RED = 1
...     GREEN = 2
...     BLUE = 3
```
What you have there are attributes that are actually types.

Reply:
The [docs for enum](https://docs.python.org/3/library/enum.html#creating-an-enum) specifically call out that there are no restrictions on the types of values used:

> Member values can be anything: int, str, etc.. If the exact value is unimportant you may use auto instances and an appropriate value will be chosen for you. Care must be taken if you mix auto with other values.

However, it seems that the only place this class is used is in looking up the appropriate class based on the string returned by `status-get`.  There are other approaches we could use to accomplish this, if the enum approach feels off.  For example, we could use [`__init_subclass__`](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__) on `Status` to register the subclasses automatically.  This would also save us from having to manually maintain the list of available statuses, although we really don't expect this list to ever change anyway.

Reply:
Note: To use the `__init_subclass__` approach, you would also need to move the ID (`active`, `blocked`, etc) up to the class level, which actually feels more natural than having it embedded in the constructors.

Reply:
Implemented this approach in #52.